### PR TITLE
Test case to export a RKE template

### PR DESCRIPTION
**Pull request by soumyalj**
@sangeethah : Please review.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_cluster_templates.py [hunk 1]
```diff
@@ -361,6 +361,159 @@ def test_cluster_template_add_readonly_member():
...
 
+def test_cluster_template_export():
+
+    # Create a DO cluster using rke config. Save a rketemplate from this
+    # cluster (with template name and revision V1).
+    # Create another cluster using the cluster template revision V1
+
+    k8sversionlist = get_k8s_versionlist()
+    standard_user_client = \
+        get_client_for_token(user_token["user_standard"]["token"])
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_cluster_templates.py]: Rename `user_standard` to reflect the fact that this is a user with create_rke_template rolebinding.
Rename `newuser_standard` to  reflect the fact that this is a standard user

File: tests/validation/tests/v3_api/test_cluster_templates.py [hunk 2]
```diff
@@ -361,6 +361,159 @@ def test_cluster_template_add_readonly_member():
...
+        name=random_test_name("revision1"),
+        clusterConfig=cluster_config1,
+        clusterTemplateId=cluster_template.id)
+
+    cluster_name = random_test_name("test-auto")
+    cluster = create_node_cluster(
+        client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision1.id)
+    check_cluster_version(cluster, k8sversionlist[0])
+
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_cluster_templates.py]: This block of code should be executed even when failure are encountered during the execution of this test.

File: tests/validation/tests/v3_api/test_cluster_templates.py [hunk 3]
```diff
@@ -361,6 +361,159 @@ def test_cluster_template_add_readonly_member():
...
+    cluster_name = random_test_name("test-auto-rkeconfig")
+    with pytest.raises(ApiError) as e:
+        create_node_cluster(standard_user_client, cluster_name,
+                            rancherKubernetesEngineConfig=rke_config)
+    print(e)
+    assert e.value.error.status == 422
+    assert e.value.error.code == "MissingRequired"
+
+    # Reset the enforcement flag to false
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_cluster_templates.py]: This block of code should be executed even when failure are encountered during the execution of this test.

File: tests/validation/tests/v3_api/test_cluster_templates.py (hunk 1 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_cluster_templates.py]: Renaming done

File: tests/validation/tests/v3_api/test_cluster_templates.py (hunk 2 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_cluster_templates.py]: Added try and finally block.

File: tests/validation/tests/v3_api/test_cluster_templates.py (hunk 3 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_cluster_templates.py]: Added try and finally block.

File: tests/validation/tests/v3_api/common.py [hunk 4]
```diff
@@ -436,8 +436,8 @@ def wait_for_pod_to_running(client, pod, timeout=DEFAULT_TIMEOUT):
     return p
 
 
-def get_schedulable_nodes(cluster):
-    client = get_user_client()
+def get_schedulable_nodes(cluster, userToken):
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/common.py]: We should default userToken parameter to USER_TOKEN

File: tests/validation/tests/v3_api/common.py [hunk 5]
```diff
@@ -449,12 +449,12 @@ def get_schedulable_nodes(cluster):
     return schedulable_nodes
 
 
-def get_role_nodes(cluster, role):
+def get_role_nodes(cluster, role, userToken):
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/common.py]: We should default userToken parameter to USER_TOKEN

File: tests/validation/tests/v3_api/common.py [hunk 6]
```diff
@@ -983,8 +984,8 @@ def wait_for_pods_in_workload(p_client, workload, pod_count,
     return pods
 
 
-def get_user_client_and_cluster():
-    client = get_user_client()
+def get_user_client_and_cluster(userToken):
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/common.py]: We should default userToken parameter to USER_TOKEN

**Comment by sangeethah** [comment]
LGTM

**Pull request merged**